### PR TITLE
Upgrade spire

### PR DESCRIPTION
**Pull request by zoemccormick**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

This PR upgrades spire server, agent and registrar service to the latest images and updates their corresponding configurations - behavior will be the same. For https://github.com/greymatter-io/greymatter/issues/426
<br/><br/>

2. What **changes to custom.yaml** are required?

None
<br/><br/>

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

<br/><br/>

To test, install in any env as usual and sidecars should get their certs as expected

### CODE REVIEW

**Review by chrisbsmith — CHANGES_REQUESTED** [review]
@zoemccormick I have not fully reviewed this, but we need to hold on all PRs against the `release-2.4` branch until we fully understand how we're going to manage the build artifacts. I think we just need to update our documentation for `1.3` to install a specific version of each artifact, but we need to verify.  Merging this PR, and another others for `1.4` will cause the CI to build a new artifact, which will let anyone install `1.4` with a simple `helm install greymatter` without providing a version number.

**Comment by zoemccormick** [comment]
@chrisbsmith ah ok thanks - I'll convert these PR's to WIP for now, and if I need to close them and reopen later I will.

File: spire/agent/values.yaml [hunk 1]
```diff
@@ -3,9 +3,11 @@ global:
     cluster_label: greymatter.io/control
   spire:
     trust_domain: quickstart.greymatter.io
```
Comment (dgoldstein1) [inline spire/agent/values.yaml:5]: do we want to update the `trust_domain` here so it's not hardcoded?

**Review by dgoldstein1 — APPROVED** [review]
lgtm! Tested on k3d

File: spire/agent/values.yaml (hunk 1 above)
Comment (chrisbsmith) [reply spire/agent/values.yaml:5]: @dgoldstein1 we do have this listed in the global.yaml as well

https://github.com/greymatter-io/helm-charts/blob/5a8c24b75c6c800174abc95b8b5575a87273fa41/global.yaml#L56

**Pull request merged**